flocker/node/_loop.py

```diff
@@ -453,8 +457,12 @@ def output_CONVERGE(self, context):
 
         with LOG_CONVERGE(self.fsm.logger, cluster_state=self.cluster_state,
                           desired_configuration=self.configuration).context():
-            d = DeferredContext(maybeDeferred(
-                self.deployer.discover_state, known_local_state))
+            d = DeferredContext(succeed(None))
+            with LOG_DISCOVERY(self.fsm.logger).context():
+                discover = DeferredContext(maybeDeferred(
+                    self.deployer.discover_state, known_local_state))
+                discover.addActionFinish()
+            d.addCallback(lambda _: discover.result)
```

Comment:
I can't say it's obvious to me how this `d` / `discover` chaining is different from just defining `d` as `discover.result` after the `with` block.  It strikes me that this might be an idiom for doing something pretty useful ... but what?

I would have like to have answered this question on my own by reading the resulting logs but since the trial logs are all mangled and unparseable and actually deploying the software would take half an hour I didn't bother.